### ISSUE TITLE
Thank you for your algorithm. I have tried it. Even non-shortest result works with tweaking

### ISSUE THREAD

**Issue by expnkx** [ulfjack/ryu#134]
![image](https://user-images.githubusercontent.com/57303909/69493348-dc6b3080-0e7b-11ea-9d0c-bacf4e604d52.png)

I think long double needs pow5_split_errors and pow5_inv_split_errors right?

**Comment by expnkx** [comment]
I have done the benchmark. Ridiculously fast. 30 times faster than FILE*. Nearly 100 time faster than std::ifstream. GREAT!
std::FILE*:     12.22499800s
std::ifstream:  31.51394400s
ibuf ryu:       0.40499600s

**Comment by ulfjack** [comment]
Be careful with non-shortest. I have a partial correctness proof, which says that the result may be off by one if the input is very close to half-way between two consecutive floats.

**Comment by ulfjack** [comment]
Yes, if you want to parse >64 bit floats, you need to use the tables that I'm using for 128 bit to shortest conversion. It is possible that the tables don't have enough bits, which I need to still double-check.

**Comment by expnkx** [comment]
> Yes, if you want to parse >64 bit floats, you need to use the tables that I'm using for 128 bit to shortest conversion. It is possible that the tables don't have enough bits, which I need to still double-check.

Ha, that is exactly why I write this as template so I can tweak it easily with different types instead of writing the same algorithm over  and over again, thanks to C++17 if constexpr.

**Comment by expnkx** [comment]
BTW, I did round for non-shortest. 4 discard. 6 carry. 5 depends on even and odd. If it is odd, then check whether it follows any non-zero number. Then it becomes the shortest one.

**Comment by ulfjack** [comment]
I'm not sure I follow. The parsing code should correctly round-trip anything output by Ryu from 64-bit double, but it's possible to manually construct cases where it doesn't return the correctly rounded value. I probably need to come up with some examples.

**Comment by expnkx** [comment]
> I'm not sure I follow. The parsing code should correctly round-trip anything output by Ryu from 64-bit double, but it's possible to manually construct cases where it doesn't return the correctly rounded value. I probably need to come up with some examples.

You can use latest GCC (needs build from source) or this GCC https://bitbucket.org/ejsvifq_mabmip/mingw-gcc/src/master/ to test my code.

**Comment by expnkx** [comment]
I have uploaded a newly compiled GCC version.
